tests/contrib/awards/conftest.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Pytest configuration.
+
+See https://pytest-invenio.readthedocs.io/ for documentation on which test
+fixtures are available.
+"""
+
+import pytest
+from invenio_indexer.api import RecordIndexer
+
+from invenio_vocabularies.contrib.awards.api import Award
+from invenio_vocabularies.contrib.awards.resources import AwardsResource, \
+    AwardsResourceConfig
+from invenio_vocabularies.contrib.awards.services import AwardsService, \
+    AwardsServiceConfig
+from invenio_vocabularies.contrib.funders.api import Funder
+
+
+@pytest.fixture(scope="module")
+def extra_entry_points():
+    """Extra entry points to load the mock_module features."""
+    return {
+        "invenio_db.models": [
+            "awards = invenio_vocabularies.contrib.awards.models",
+        ],
+        "invenio_jsonschemas.schemas": [
+            "awards = \
+                invenio_vocabularies.contrib.awards.jsonschemas",
+            "funders = \
+                invenio_vocabularies.contrib.funders.jsonschemas"
+        ],
+        "invenio_search.mappings": [
+            "awards = \
+                invenio_vocabularies.contrib.awards.mappings",
+        ]
+    }
+
+
+@pytest.fixture(scope="function")
+def award_full_data():
+    """Full award data."""
+    return {
+        "pid": "755021",
+        "identifiers": [
+            {
+                "identifier": "https://cordis.europa.eu/project/id/755021",
+                "scheme": "url"
+            }
+        ],
+        "number": "755021",
+        "title": {
+            "en": "Personalised Treatment For Cystic Fibrosis Patients With \
+                Ultra-rare CFTR Mutations (and beyond)",
+        }
+    }
+
+
+@pytest.fixture(scope="function")
+def example_award(db, award_full_data, identity, indexer, service):
+    """Creates and hard deletes an award."""
+    award = service.create(identity, award_full_data)
+    Award.index.refresh()  # Refresh the index
+    yield award
+    award._record.delete(force=True)
+    indexer.delete(award._record, refresh=True)
+    db.session.commit()
+
+
+@pytest.fixture(scope="function")
+def example_funder(db):
+    """Example funder."""
+    api_funder = {
+        "name": "CERN",
+        "title": {
+            "en": "European Organization for Nuclear Research",
+            "fr": "Organisation européenne pour la recherche nucléaire"
+        },
+        "country": "CH"
+    }
+    # at API level 'pid' is passed as an arg
+    fun = Funder.create(api_funder, pid="01ggx4157")
+    fun.commit()
+    db.session.commit()
+    return fun
```

Comment:
minor: won't we have the same problem here with the soft deletion? (i.e. we have to yield + hard delete?)

Reply:
The problem wasn't appearing because the fixture was not used more than once in the same module 😅 I added the deletion.